webapp/src/utils/jira_issue_metadata.tsx

```diff
@@ -224,6 +228,44 @@ export function getCustomFieldFiltersForProjects(metadata: IssueMetadata | null,
         } as FilterField);
     }
 
+    // Filtering out the statuses on the basis of selected issue types
+    const issueTypesWithStatuses = metadata && metadata.issue_types;
+    const keys = new Set<string>();
+    const statuses: Status[] = [];
```

Comment:
I think this and code below could be extracted into a separate function. It could return `statusField | undefined`, then this function could conditionally add it to `result` if it's not `undefined`.

We should also edit the existing tests in [jira_issue_metadata.test.tsx](https://github.com/mattermost/mattermost-plugin-jira/blob/master/webapp/src/utils/jira_issue_metadata.test.tsx) to include checks for statuses

---

I wrote the below comment before I noticed that there are indeed tests for this file:

It would be great to have `getCustomFieldFiltersForProjects` or others in this file unit tested but not necessary right now. This is the sort of task that potentially makes a good HW ticket. If you see an potential improvement to the code such as code missing unit tests, I encourage you to create a ticket to document your thought and put the "Help Wanted" etc. labels on the ticket. "Write tests for selectors in jira_issue_metadata.tsx`